From 98680a85df888f86e56ec12d78b2de730426e5af Mon Sep 17 00:00:00 2001
From: Randy Dunlap <rdunlap@infradead.org>
Date: Sun, 19 Jul 2026 20:12:36 -0700
Subject: [PATCH 06004/15333] posix-cpu-timers: Avoid kernel-doc warnings

Use the struct keyword when describing a struct. Without that, the
type defaults to function.

Warning: ./include/linux/posix-timers_types.h:40 cannot understand
 function prototype: 'struct posix_cputimer_base'
Warning: ./include/linux/posix-timers_types.h:55 cannot understand
 function prototype: 'struct posix_cputimers'
Warning: ./include/linux/posix-timers_types.h:67 cannot understand
 function prototype: 'struct posix_cputimers_work'

Signed-off-by: Randy Dunlap <rdunlap@infradead.org>
Signed-off-by: Thomas Gleixner <tglx@kernel.org>
Link: https://patch.msgid.link/20260720031236.3070575-2-rdunlap@infradead.org
---
 include/linux/posix-timers_types.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/linux/posix-timers_types.h b/include/linux/posix-timers_types.h
index a4712c1008c9..b40cf352e01d 100644
--- a/include/linux/posix-timers_types.h
+++ b/include/linux/posix-timers_types.h
@@ -34,7 +34,7 @@
 #ifdef CONFIG_POSIX_TIMERS
 
 /**
- * posix_cputimer_base - Container per posix CPU clock
+ * struct posix_cputimer_base - Container per posix CPU clock
  * @nextevt:		Earliest-expiration cache
  * @tqhead:		timerqueue head for cpu_timers
  */
@@ -44,7 +44,7 @@ struct posix_cputimer_base {
 };
 
 /**
- * posix_cputimers - Container for posix CPU timer related data
+ * struct posix_cputimers - Container for posix CPU timer related data
  * @bases:		Base container for posix CPU clocks
  * @timers_active:	Timers are queued.
  * @expiry_active:	Timer expiry is active. Used for
@@ -60,7 +60,7 @@ struct posix_cputimers {
 };
 
 /**
- * posix_cputimers_work - Container for task work based posix CPU timer expiry
+ * struct posix_cputimers_work - Container for task work based posix CPU timer expiry
  * @work:	The task work to be scheduled
  * @mutex:	Mutex held around expiry in context of this task work
  * @scheduled:  @work has been scheduled already, no further processing
-- 
2.55.0

